src/diagnostics/core_diagnostics.jl

```diff
@@ -273,10 +273,39 @@ add_diagnostic_variable!(
     or from mixing length closure with EDMF SGS model.
     """,
     compute! = (out, state, cache, time) -> begin
+        turbconv_model = cache.atmos.turbconv_model
+        # TODO: consolidate remaining mixing length types
+        # (smagorinsky_lilly, dz) into a single mixing length function
+        if isa(turbconv_model, PrognosticEDMFX) ||
+           isa(turbconv_model, DiagnosticEDMFX) ||
+           isa(turbconv_model, EDOnlyEDMFX)
+            ᶜmixing_length = mixing_length(state, cache)
+        else
+            (; params) = cache
+            (; ᶜlinear_buoygrad, ᶜstrain_rate_norm) = cache.precomputed
+            ᶜdz = Fields.Δz_field(axes(state.c))
+            ᶜprandtl_nvec = cache.scratch.ᶜtemp_scalar_2
+            @. ᶜprandtl_nvec = turbulent_prandtl_number(
+                params,
+                ᶜlinear_buoygrad,
+                ᶜstrain_rate_norm,
+            )
+            ᶜmixing_length = @. lazy(
```

Comment:
Shouldn't this one be materialized, if we actually want to see the values in the output?

Reply:
This is materialized a few lines down (`out .= ᶜmixing_length` or `copy(ᶜmixing_length)`)